autopilot/scanner/scanner.go

```diff
@@ -108,14 +125,16 @@ func (s *scanner) Scan(ctx context.Context, hs HostScanner, force bool) {
 
 	s.wg.Add(1)
 	go func() {
-		defer s.wg.Done()
-
+		defer func() {
+			s.mu.Lock()
+			s.scanning = false
```

Comment:
Do we still need `scanning` as a field? If we always call `cancel` after we are done scanning, we could 
```go
select {
case <-ctx.Done():
  return false
default:
  return true
```
in a `scanning()` helper to determine whether we are actually scanning atm.